tests/test_ec2.py

```diff
@@ -205,6 +205,24 @@ def test_ec2_resize(self):
             list(sorted(['m4.large', 'm4.2xlarge', 'm4.2xlarge'])))
 
 
+class TestStateTransitionAgeFilter(BaseTest):
+
+    def test_ec2_state_transition_age(self):
+        session_factory = self.replay_flight_data(
+            'test_ec2_state_transition_age_filter'
+        )
+        policy = self.load_policy({
+            'name': 'ec2-state-transition-age',
+            'resource': 'ec2',
+            'filters': [
+                {'State.Name': 'running'},
+                {'type': 'state-age',
+                 'days': 30}]},
+            session_factory=session_factory)
+        resources = policy.run()
+        self.assertEqual(len(resources), 1)
```

Comment:
it would be good to do an additional verify that the resource found has the attribute/condition that is being sought after. in this case it looks like there are three instances in the describe, two of which satisfy the condition which seems odd that only one would be returned. a direct equality match on state transition reason would suffice for the result validation.